language-plutus-core/generators/Language/PlutusCore/Generators/Interesting.hs

```diff
@@ -51,21 +52,17 @@ type TermGen uni a = Gen (TermOf uni a)
 --
 -- > ifThenElse {integer -> integer -> integer} (lessThanInteger i j) addInteger subtractInteger i j
 -- >     == if i < j then i + j else i - j
-genOverapplication :: Generatable uni => TermGen uni Integer
+genOverapplication :: Generatable uni => TermGen uni Integer  -- FIXME: Doesn't make sense any more???
```

Comment:
Why not? The stdlib is mostly there for tests, for example (apart from the `Type.hs` module, n-ary tuples and maybe a few more things).

Reply:
> I think @michaelpj also eta-expands builtins in Plutus Tx. Is that right?

Yes, but we're hoping that will go away: see https://jira.iohk.io/browse/SCP-480.

When I said "we don't want to use it in general" I was worried about the possibility of ending up with inefficient wrapped builtin applications in places where you could just have a direct application, and we should be trying to generate those wherever possible (and up front, rather than as a later optimisation).  What happens in the PlutusTx output is a different matter: I think it's OK have a later optimisation phase there.